app/models/host/base.rb

```diff
@@ -175,7 +173,8 @@ def set_interfaces(parser)
       end
 
       parser.interfaces.each do |name, attributes|
-        iface = get_interface_scope(name, attributes).first || interface_class(name).new(:managed => false)
+        iface_scope = get_interface_scope(name, attributes)
+        iface = iface_scope.nil? ? interface_class(name).new(:managed => false) : iface_scope.first
```

Comment:
This removes important handling of `iface_scope.first` being nil, hence `iface` remains nil.